backoff/_wait_gen.py

```diff
@@ -5,10 +5,10 @@
 
 
 def expo(
-    base: int = 2,
-    factor: int = 1,
-    max_value: Optional[int] = None
-) -> Generator[int, Any, None]:
+    base: float = 2,
+    factor: float = 1,
+    max_value: Optional[float] = None
+) -> Generator[Optional[float], Any, None]:
```

Comment:
This unfortunately broke the typing for `_WaitGenerator`. I.e. you can no longer do `backoff.on_exception(backoff.expo, ...)` without a type error, in v2.2.0.

https://github.com/litl/backoff/blob/9a20a7fe9fc13621ab966f151d6324649bd1e814/backoff/_typing.py#L44

/CC: @bgreen-litl

Reply:
see https://github.com/litl/backoff/issues/177

Reply:
@cdce8p @calpaterson Should be fixed in backoff 2.2.1. Let me know if you find otherwise